### PR TITLE
Add Javadoc documentation to all classes and interfaces.

### PR DESCRIPTION
**Pull request by mjuvekar7**
Add Javadoc documentation to all classes and interfaces of the Fluid module.

### CODE REVIEW

File: src/main/java/org/terasology/fluid/component/FluidContainerItemComponent.java [hunk 1]
```diff
@@ -22,18 +22,31 @@
 import org.terasology.rendering.assets.texture.TextureRegionAsset;
 
 /**
- * @author Marcin Sciesinski <marcins78@gmail.com>
+ * This component indicates that an entity is a container capable of holding fluids.
  */
 public class FluidContainerItemComponent implements Component, ItemDifferentiating {
+    /** The type of the fluid it can contain */
     @Replicate
     public String fluidType;
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/component/FluidContainerItemComponent.java:31]: Need a newline after every field here, i.e.

```java
/** The type of the fluid it can contain */
@Replicate
public String fluidType;

/** The current volume of fluid in the container */
public float volume;

/** The maximum volume ... */
```

File: src/main/java/org/terasology/fluid/component/FluidContainerItemComponent.java [hunk 2]
```diff
@@ -22,18 +22,31 @@
...
+    /** The texture of the container when it is completely filled */
     public TextureRegionAsset<?> textureWithHole;
+    /** The texture of the container when it is empty */
     public TextureRegionAsset<?> emptyTexture;
 
+    /**
+     * Checks whether the fluid container's attributes are the same as those of a given object.
+     *
+     * @param o the Object to compare the fluid container to
+     * @return (boolean) whether the fluid container's attributes are the same as that of the given object
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/component/FluidContainerItemComponent.java]: `(boolean)` isn't necessary here - that's something dynamically typed languages do. We know what the return type is 🙂

File: src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java [hunk 3]
```diff
@@ -18,31 +18,67 @@
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.AbstractConsumableEvent;
 
+/**
+ * This event indicates that an entity is putting a fluid in an inventory slot, and contains attributes indicating the
+ * state of the fluid before it was put in the inventory.
+ */
 public class BeforeFluidPutInInventory extends AbstractConsumableEvent {
+    /** The entity who is putting the fluid in the inventory */
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java]: "the entity who is putting" → "the entity which is placing"

File: src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java [hunk 4]
```diff
@@ -18,31 +18,67 @@
...
     private EntityRef instigator;
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java:28]: Need a newline after every field here [2]

File: src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java [hunk 5]
```diff
@@ -18,31 +18,67 @@
...
     private String fluidType;
+    /** The volume of the fluid being put in the inventory */
     private float volume;
+    /** The slot number of the inventory slot in which the fluid is being put */
     private int slot;
 
+    /**
+     * Parametrized constructor.
+     *
+     * @param instigator the entity who is putting the fluid in the inventory
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java]: who → which, putting → placing

File: src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java [hunk 6]
```diff
@@ -18,31 +18,67 @@
...
+     * @return the fluid type of the fluid being put in the inventory
+     */
     public String getFluidType() {
         return fluidType;
     }
 
+    /**
+     * Accessor function that returns the instigator of the action.
+     *
+     * @return an EntityRef to the entity who is putting the fluid in the inventory
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/event/BeforeFluidPutInInventory.java]: who → which, putting → placing. There are a few more of these so I won't list them all - do a ctrl+f for "who" and "put", replace where necessary

File: src/main/java/org/terasology/fluid/event/BeforeFluidRemovedFromInventory.java [hunk 7]
```diff
@@ -18,31 +18,67 @@
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.AbstractConsumableEvent;
 
+/**
+ * This event indicates that an entity is removing a fluid in from inventory slot, and contains attributes indicating
+ * the state of the fluid before it was removed from the inventory.
+ */
 public class BeforeFluidRemovedFromInventory extends AbstractConsumableEvent {
+    /** The entity who is removing the fluid from the inventory */
     private EntityRef instigator;
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/event/BeforeFluidRemovedFromInventory.java:28]: Need a newline after every field here [3]

File: src/main/java/org/terasology/fluid/event/FluidVolumeChangedInInventory.java [hunk 8]
```diff
@@ -18,13 +18,30 @@
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.Event;
 
+/**
+ * This event indicates that the volume of a fluid was changed while it was in an inventory.
+ */
 public class FluidVolumeChangedInInventory implements Event {
+    /** The instigator of the action */
     private EntityRef instigator;
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/event/FluidVolumeChangedInInventory.java:27]: Need a newline after every field here [4]

File: src/main/java/org/terasology/fluid/component/FluidContainerItemComponent.java [hunk 9]
```diff
@@ -22,18 +22,31 @@
...
     public Vector2f fluidSizePerc;
+    /** The texture of the container when it is completely filled */
     public TextureRegionAsset<?> textureWithHole;
+    /** The texture of the container when it is empty */
     public TextureRegionAsset<?> emptyTexture;
 
+    /**
+     * Checks whether the fluid container's attributes are the same as those of a given object.
+     *
+     * @param o the Object to compare the fluid container to
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/component/FluidContainerItemComponent.java]: Every comment of the type "the X that does Y" needs to be changed to "The X that does Y." At one point you're actually doing the reverse - that should definitely be undone 😕

File: src/main/java/org/terasology/fluid/system/FluidManager.java [hunk 10]
```diff
@@ -17,71 +17,74 @@
...
 import org.terasology.entitySystem.entity.EntityRef;
 
+/**
+ * Interface for a generic fluid manager.
+ */
 public interface FluidManager {
     /**
      * Add a certain volume of fluid to all fluid inventory slots.
      *
-     * @param instigator    The entity that's instigating this action.
```
Comment (rzats) [inline src/main/java/org/terasology/fluid/system/FluidManager.java:24]: As mentioned above, this is actually the correct way to do parameter javadocs. Undo this change, please 🙂

**Review by rzats — CHANGES_REQUESTED** [review]
Thanks for the contribution! Needs a few changes - some minor, others fairly significant.

File: src/main/java/org/terasology/fluid/system/FluidManager.java (hunk 10 above)
Comment (rzats) [reply src/main/java/org/terasology/fluid/system/FluidManager.java:24]: (here and everywhere else, that is)

**Comment by Cervator** [comment]
Thanks for the nice javadoc! :-)

(@rzats please note if there was anything left you expected to be fixed)
